libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -363,10 +363,11 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
     }
 
     // when landing, add half of head-wind.
-    const int32_t head_wind_compensation_cm = head_wind() * 0.5f * 100;
+    const float head_wind_comp = constrain_float((float)wind_comp, 0.0f, 100.0f)*0.01;
+    const int32_t head_wind_compensation_cm = head_wind() * head_wind_comp * 100;
 
-    // Do not lower it or exceed cruise speed
-    return constrain_int32(target_airspeed_cm + head_wind_compensation_cm, target_airspeed_cm, aparm.airspeed_cruise_cm);
+    // Do not lower it or exceed max airspeed `ARSPD_FBW_MAX`
+    return constrain_int32(target_airspeed_cm + head_wind_compensation_cm, target_airspeed_cm, aparm.airspeed_max*100);
```

Comment:
this change would need to be enabled with an OPTIONS bit from LAND_OPTIONS

Reply:
please add a option_enabled() method and use for OptionsMask::ON_LANDING_FLARE_USE_THR_MIN

Reply:
complete